test-data/unit/check-classes.test

```diff
@@ -1665,6 +1665,83 @@ a.f = a.f # E: Property "f" defined in "A" is read-only
 a.f.x # E: "int" has no attribute "x"
 [builtins fixtures/property.pyi]
 
+[case testPropertyWithTupleUnionAndNonUnionMismatchReturnValueType]
+from typing import Tuple, Union
+class A:
+    a: str
+    b: str
+    c: str
+    d: str
+    e: str
+    f: str
+    g: Union[str, int]
+    h: Union[str, float]
+    i: Union[str, None]
+    j: Union[str, None]
+    k: Union[str, None]
+    l: Union[str, None]
+
+    @property
+    def x(self) -> Tuple[str, str, str, str, str, str, str, str, str, str, str, str]:
+        return (
+            self.a,
+            self.b,
+            self.c,
+            self.d,
+            self.e,
+            self.f,
+            self.g,
+            self.h,
+            self.i,
+            self.j,
+            self.k,
+            self.l,
+        )
+[out]
+main:18: error: Incompatible return value type (6 tuple items are incompatible; 3 items are omitted)
+main:18: note: Expression tuple item 6 has type "Union[str, int]"; "str" expected;
+main:18: note: Expression tuple item 7 has type "Union[str, float]"; "str" expected;
+main:18: note: Expression tuple item 8 has type "Optional[str]"; "str" expected;
+[builtins fixtures/property.pyi]
+
+[case testPropertyWithTupleUnionAndNonUnionMismatchReturnValueType_wider_expeced_return]
+from typing import Tuple, Union
+class A:
```

Comment:
I'd really suggest eliminating the class and writing a simple module-level assignment (so that this code can be moved to `check-tuples.test`) but that's also mostly a style nit.

Reply:
Sorry, that comment didn't load before I refreshed the page - I'm fine with keeping the class version:) Thanks!